components/provisioner/internal/provisioning/service.go

```diff
@@ -475,6 +459,28 @@ func (r *service) RollBackLastUpgrade(runtimeID string) (*gqlschema.RuntimeStatu
 	return r.RuntimeStatus(runtimeID)
 }
 
+func (r *service) isKyma2Installed(cluster model.Cluster) bool {
```

Comment:
I'm not sure about the name of this function. There is for example a condition with the existence of the kubeconfig which is not strictly related to the Kyma1.X/ Kyma2.X thing. Maybe we could go for something like `shouldDeprovisionWithoutUninstall`